subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/constraints/PublishedDependencyConstraintsIntegrationTest.groovy

```diff
@@ -541,4 +542,100 @@ dependencies {
         }
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/24037")
+    @RequiredFeature(feature = GradleMetadataResolveRunner.GRADLE_METADATA, value="true")
+    void "duplicate constraint going back to pending not leave hanging edge"() {
```

Comment:
"Pending dependencies" is how we name discovered constraints that are not active because there is no concrete dependency to that module. It is used in code, see `PendingDependenciesVisitor`, and test names in multiple places.